modules/akka-http/src/main/scala/pureconfig/module/akkahttp/package.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig.module
+
+import akka.http.scaladsl.model.{IllegalUriException, Uri}
+import akka.http.scaladsl.model.Uri.ParsingMode
+import pureconfig.{ConfigReader, ConfigWriter}
+import pureconfig.error.CannotConvert
+
+import scala.util.Try
+
+package object akkahttp {
+
+  implicit val uriReader: ConfigReader[Uri] =
+    ConfigReader.fromString(str =>
+      Try(Uri(str, ParsingMode.Strict)).fold(
```

Comment:
nit: you can use `Try(...).toEither.left.map(...)` to simplify this expression, by getting rid of the success `case` and the need to wrap errors in `Left`.